crates/sui-core/src/authority/authority_store.rs

```diff
@@ -62,52 +69,95 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, const USE_LOCKS: bool, S> {
+    ///   successfully, this map must be up to date. If the RPC fails before returning, the client
+    ///   retries, which will eventually cause this map to be up to date.
     transactions: DBMap<TransactionDigest, TransactionEnvelope<S>>,
 
     /// This is a map between the transaction digest and the corresponding certificate for all
     /// certificates that have been successfully processed by this authority. This set of certificates
     /// along with the genesis allows the reconstruction of all other state, and a full sync to this
     /// authority.
+    ///
+    /// Consistency requirements:
+    /// - Read-after-write with respect to Validator::confirmation_transaction
+    /// - A certificate must be added to this map *before* the certificate's input object
+    ///   locks are deleted and the output object locks are initialized
     certificates: DBMap<TransactionDigest, CertifiedTransaction>,
 
     /// The map between the object ref of objects processed at all versions and the transaction
     /// digest of the certificate that lead to the creation of this version of the object.
     ///
     /// When an object is deleted we include an entry into this table for its next version and
     /// a digest of ObjectDigest::deleted(), along with a link to the transaction that deleted it.
+    ///
+    /// Consistency requirements:
+    /// - Read-after-write with respect to Validator::confirmation_transaction
     parent_sync: DBMap<ObjectRef, TransactionDigest>,
 
     /// A map between the transaction digest of a certificate that was successfully processed
     /// (ie in `certificates`) and the effects its execution has on the authority state. This
     /// structure is used to ensure we do not double process a certificate, and that we can return
     /// the same response for any call after the first (ie. make certificate processing idempotent).
+    ///
+    /// Consistency requirements:
+    /// - Same as for `certificates`
     effects: DBMap<TransactionDigest, TransactionEffectsEnvelope<S>>,
 
+    // The following three tables (sequenced, schedule, last_consensus_index) are critical for
+    // sequencing shared object txes correctly, and must be updated atomically.
+
     /// Hold the lock for shared objects. These locks are written by a single task: upon receiving a valid
     /// certified transaction from consensus, the authority assigns a lock to each shared objects of the
-    /// transaction. Note that all authorities are guaranteed to assign the same lock to these objects.
+    /// transaction. Note that all authorities are guaranteed to assign the same lock to these objects,
+    /// because shared object certs are received from narwhal in the same order by all authorities,
+    /// and are processed serially in a single task.
+    ///
+    /// sequenced contains a map form (TransactionDigest, ObjectID) to the object version number
+    /// that will be created when that Transaction is sequenced.
```

Comment:
> Entries in schedule needs to be kept while the shared object is live.

Sorry can you elaborate what that means - the shared object is live?

Reply:
I need to schedule a call with Alberto to make sure I understand how these work, but currently they are both cleaned up in `update_state` at the same time, and there is a comment that these should be combined into a single table. So, I think these have the same longevity requirements. `schedule` just seems to exist so that we can find the next available version for the shared object.

Reply:
> Sorry can you elaborate what that means - the shared object is live

If we can delete shared objects, then we should keep this entry while the object is not deleted.

Reply:
@gdanezis that could be a really really long time....